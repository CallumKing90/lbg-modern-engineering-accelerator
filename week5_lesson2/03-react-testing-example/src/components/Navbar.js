import React from 'react'

const Navbar = ({ onNavChange }) => (
  <nav>
    <ul>
      <li>
        <button onClick={() => onNavChange('Home')}>Home</button>
      </li>
      <li>
        <button onClick={() => onNavChange('About')}>About</button>
      </li>
      <li>
        <button onClick={() => onNavChange('Contact')}>Contact</button>
      </li>
      <li>
        <button onClick={() => onNavChange('StarWars')}>Star Wars</button>
      </li>
    </ul>
  </nav>
)

export default Navbar
