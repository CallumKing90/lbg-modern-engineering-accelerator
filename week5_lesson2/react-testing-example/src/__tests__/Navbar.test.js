import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Navbar from '../components/Navbar';

describe('<Navbar />', () => {
  it('calls onNavChange when a button is clicked', () => {
    const mockOnNavChange = jest.fn();
    const component = render(<Navbar onNavChange={mockOnNavChange} />);

    const homeButton = component.getByText('Home');
    fireEvent.click(homeButton);

    expect(mockOnNavChange).toHaveBeenCalledTimes(1);
    expect(mockOnNavChange).toHaveBeenCalledWith('Home');
  });

  it('has 4 items', () => {
    const { getByText } = render(<Navbar onNavChange={() => {}} />);
    expect(getByText('Home')).toBeInTheDocument();
    expect(getByText('About')).toBeInTheDocument();
    expect(getByText('Contact')).toBeInTheDocument();
    expect(getByText('Star Wars')).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const mockOnNavChange = jest.fn();
    const { baseElement } = render(<Navbar onNavChange={mockOnNavChange} />);
    expect(baseElement).toMatchSnapshot();
  });
});
