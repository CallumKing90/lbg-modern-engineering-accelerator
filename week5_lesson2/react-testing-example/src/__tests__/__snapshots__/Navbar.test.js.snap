// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> matches the snapshot 1`] = `
<body>
  <div>
    <nav>
      <ul>
        <li>
          <button>
            Home
          </button>
        </li>
        <li>
          <button>
            About
          </button>
        </li>
        <li>
          <button>
            Contact
          </button>
        </li>
        <li>
          <button>
            Star Wars
          </button>
        </li>
      </ul>
    </nav>
  </div>
</body>
`;
